src/Process/3dTilesProcessing.js

```diff
@@ -0,0 +1,174 @@
+import * as THREE from 'three';
+import Fetcher from '../Core/Scheduler/Providers/Fetcher';
+import { $3dTilesIndex } from '../Core/Scheduler/Providers/3dTiles_Provider';
+
+function requestNewTile(view, scheduler, geometryLayer, metadata, parent) {
+    const command = {
+        /* mandatory */
+        view,
+        requester: parent,
+        layer: geometryLayer,
+        priority: 10000,
+        /* specific params */
+        metadata,
+        redraw: false,
+    };
+
+    return scheduler.execute(command);
+}
+
+function subdivideNode(context, layer, node) {
+    if (!node.pendingSubdivision && node.children.filter(n => n.layer == layer.id).length == 0) {
+        node.pendingSubdivision = true;
+
+        const childrenTiles = layer.tileIndex.index[node.tileId].children;
+        if (childrenTiles === undefined) {
+            return;
+        }
+
+        const promises = [];
+        for (let i = 0; i < childrenTiles.length; i++) {
+            promises.push(
+                requestNewTile(context.view, context.scheduler, layer, childrenTiles[i], node).then((tile) => {
+                    node.add(tile);
+                    tile.updateMatrixWorld();
+                    if (node.additiveRefinement) {
+                        context.view.notifyChange(0, true);
+                    }
+                }));
+        }
+
+        Promise.all(promises).then(() => {
+            node.pendingSubdivision = false;
+            context.view.notifyChange(0, true);
+        });
+    }
+}
+
+export function $3dTilesCulling(node, camera) {
+    if (node.boundingVolume.region) {
+        // TODO
+    }
+    if (node.boundingVolume.box) {
+        return !camera.isBox3DVisible(node.boundingVolume.box, node.matrixWorld);
+    } else if (node.boundingVolume.sphere) {
+        // TODO return !camera.isSphereVisible(node.boundingVolume.sphere, node.matrixWorld);
+    }
+    return false;
+}
+
+let preSSE;
+export function pre3dTilesUpdate(context) {
+    // pre-sse
+    const hypotenuse = Math.sqrt(context.camera.width * context.camera.width + context.camera.height * context.camera.height);
+    const radAngle = context.camera.FOV * Math.PI / 180;
+
+     // TODO: not correct -> see new preSSE
+    // const HFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) / context.camera.ratio);
+    const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.width);
+    preSSE = hypotenuse * (2.0 * Math.tan(HYFOV * 0.5));
+}
+
+function computeNodeSSE(camera, node) {
+    if (node.boundingVolume.region) {
+        // TODO
+    }
+    if (node.boundingVolume.box) {
+        // TODO: compute proper distance
+        const worldCoordinateCenter = node.boundingVolume.box.getCenter();
+        worldCoordinateCenter.applyMatrix4(node.matrixWorld);
+        const distance = camera.camera3D.position.distanceTo(worldCoordinateCenter);
+        return preSSE * (node.geometricError / distance);
```

Comment:
I also don't understand this. Could you please explain how you got to that expression? The [calculation](https://github.com/AnalyticalGraphicsInc/cesium/blob/38dab936cff23e8f99fb79a3873adfe9400d3bab/Source/Scene/QuadtreePrimitive.js#L688) in Cesium makes more sense to me, but I'd like to understand your reasoning. Thanks.